src/main/java/com/gotocompany/depot/http/request/util/JsonParserUtils.java

```diff
@@ -0,0 +1,115 @@
+import com.gotocompany.depot.exception.InvalidTemplateException;
+import com.gotocompany.depot.message.ParsedMessage;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.Map;
+
+
+public class JsonParserUtils {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper()
+            .enable(DeserializationFeature.FAIL_ON_TRAILING_TOKENS);
+
+    public static JsonNode createJsonNode(String jsonTemplate) {
+        if (jsonTemplate.isEmpty()) {
+            throw new ConfigurationException("Json body template cannot be empty");
+        }
+        try {
+            return OBJECT_MAPPER.readTree(jsonTemplate);
+
+        } catch (JsonSyntaxException | IOException e) {
+            throw new ConfigurationException(String.format("Json body template is not a valid json. %s", e.getMessage()));
+        }
+    }
+
+    public static JsonNode parse(JsonNode jsonNode, ParsedMessage parsedMessage) {
+        switch (jsonNode.getNodeType()) {
+            case ARRAY:
+                return parseInternal((ArrayNode) jsonNode, parsedMessage);
+            case OBJECT:
+                return parseInternal((ObjectNode) jsonNode, parsedMessage);
+            case STRING:
+                return parseInternal((TextNode) jsonNode, parsedMessage);
+            case NUMBER:
+            case BOOLEAN:
+            case NULL:
+                return parseInternal(jsonNode, parsedMessage);
+            default:
+                throw new IllegalArgumentException("The provided Json type is not supported");
+        }
+    }
+
+    public static JsonNode parseInternal(ObjectNode objectNode, ParsedMessage parsedMessage) {
+        ObjectNode finalJsonObject = new JsonNodeFactory(false).objectNode();
+        for (Map.Entry<String, JsonNode> entry : objectNode.properties()) {
+            String keyString = entry.getKey();
+            TextNode keyStringNode = new TextNode(keyString);
+            JsonNode parsedKeyNode = parseInternal(keyStringNode, parsedMessage);
+            String parsedKeyString = parsedKeyNode.toString();
+            if (parsedKeyNode.getNodeType().equals(JsonNodeType.STRING)) {
+                parsedKeyString = parsedKeyString.substring(1, parsedKeyString.length() - 1);
+            }
+            JsonNode valueNode = entry.getValue();
+            JsonNode parsedValue = JsonParserUtils.parse(valueNode, parsedMessage);
+            finalJsonObject.put(parsedKeyString, parsedValue);
+        }
+        return finalJsonObject;
+    }
+
+
+    public static JsonNode parseInternal(ArrayNode arrayNode, ParsedMessage parsedMessage) {
+        ArrayNode tempJsonArray = new JsonNodeFactory(false).arrayNode();
+        for (Iterator<JsonNode> it = arrayNode.elements(); it.hasNext();) {
+            JsonNode jsonElement1 = it.next();
+            JsonNode parsedJsonNode = JsonParserUtils.parse(jsonElement1, parsedMessage);
+            tempJsonArray.add(parsedJsonNode);
+        }
+        return tempJsonArray;
```

Comment:
I think we can rewrite this with below lines; Iterator way is not much expressive.

```
        ArrayNode parsedArrayNode = new JsonNodeFactory(false).arrayNode();
        arrayNode.forEach(jsonNode -> parsedArrayNode.add(parse(jsonNode, parsedMessage)));
        return parsedArrayNode;
```

Reply:
Hmmm this one
```
        ArrayNode parsedArrayNode = JsonNodeFactory.instance.arrayNode();
        arrayNode.forEach(jsonNode -> parsedArrayNode.add(parse(jsonNode, parsedMessage)));
        return parsedArrayNode;
```

Reply:
fixed